packages/typespec-ts/test/azureIntegration/azureArmResources.spec.ts

```diff
@@ -454,4 +544,479 @@ describe("Azure Arm Resources Rest Client", () => {
+    assert.deepStrictEqual(resourceGroupResult, [validResourceGroupExtensionsResource]);
+
+    const subscriptionResponse = await client
+      .path(
+        "/{resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources",
+        {
+          value: `subscriptions/${SUBSCRIPTION_ID_EXPECTED}`,
+          allowReserved: true
+        }
+      )
+      .get();
+    if (isUnexpected(subscriptionResponse)) {
+      const error = `Unexpected status code ${subscriptionResponse.status}`;
+      assert.fail(error);
+    }
+    const subscriptionIter = paginate(client, subscriptionResponse);
+    let subscriptionIterResult = [];
+    for await (const item of subscriptionIter) {
+      subscriptionIterResult.push(item);
+    }
+    assert.strictEqual(subscriptionResponse.status, "200");
+    assert.deepStrictEqual(subscriptionIterResult, [validSubscriptionExtensionsResource]);
+
+    const tenantResponse = await client
+      .path(
+        "/{resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources",
+        {
+          value: "",
+          allowReserved: true
+        }
+      )
+      .get();
+    if (isUnexpected(tenantResponse)) {
+      const error = `Unexpected status code ${tenantResponse.status}`;
+      assert.fail(error);
+    }
+    const tenantIter = paginate(client, tenantResponse);
+    let tenantResult = [];
+    for await (const item of tenantIter) {
+      tenantResult.push(item);
+    }
+    assert.strictEqual(tenantResponse.status, "200");
+    assert.deepStrictEqual(tenantResult, [validTenantExtensionsResource]);
+
+    const resourceResponse = await client
+      .path(
+        "/{resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources",
+        {
+          value: `subscriptions/${SUBSCRIPTION_ID_EXPECTED}/resourceGroups/${RESOURCE_GROUP_EXPECTED}/providers/Azure.ResourceManager.Resources/topLevelTrackedResources/top`,
+          allowReserved: true
+        }
+      )
+      .get();
+    if (isUnexpected(resourceResponse)) {
+      const error = `Unexpected status code ${resourceResponse.status}`;
+      assert.fail(error);
+    }
+    const resourceIter = paginate(client, resourceResponse);
+    let resourceResult = [];
+    for await (const item of resourceIter) {
+      resourceResult.push(item);
+    }
+    assert.strictEqual(resourceResponse.status, "200");
+    assert.deepStrictEqual(resourceResult, [validResourceExtensionsResource]);
+  });
+
+  it("Set allowReserved false, expect error", async () => {
+    try {
+      await client.path(
+        "/{resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources/{extensionsResourceName}",
+        {
+          value: `subscriptions/${SUBSCRIPTION_ID_EXPECTED}/resourceGroups/${RESOURCE_GROUP_EXPECTED}`,
+          allowReserved: false as any
+        },
+        "extension"
+      )
+        .get();
+      assert.fail("Set allowReserved false, expect error");
+    } catch (error) {
```

Comment:
404 error?

Reply:
client:
![image](https://github.com/user-attachments/assets/4939a89a-c721-442b-8ab7-394f16750116)
server: 
![image](https://github.com/user-attachments/assets/9c369d41-a68f-4ba8-a207-88c0dd674f8e)

Reply:
This is not expected.